roborock/containers.py

```diff
@@ -261,12 +249,60 @@ class Status(RoborockBase):
     unsave_map_reason: Optional[int] = None
     unsave_map_flag: Optional[int] = None
 
-    def update_status(self, model_specification: ModelSpecification) -> None:
-        self.fan_power_enum = model_specification.fan_power_code.as_enum_dict()[self.fan_power]
-        if model_specification.mop_mode_code is not None:
-            self.mop_mode_enum = model_specification.mop_mode_code.as_enum_dict()[self.mop_mode]
-        if model_specification.mop_intensity_code is not None:
-            self.water_box_mode_enum = model_specification.mop_intensity_code.as_enum_dict()[self.water_box_mode]
+
+@dataclass
+class S5MaxStatus(Status):
+    fan_power: Optional[RoborockFanSpeedS6Pure] = None
+    water_box_mode: Optional[RoborockMopIntensityV2] = None
+
+
+@dataclass
+class Q7MaxStatus(Status):
+    fan_power: Optional[RoborockFanSpeedQ7Max] = None
+    water_box_mode: Optional[RoborockMopIntensityV2] = None
+
+
+@dataclass
+class S6MaxVStatus(Status):
+    fan_power: Optional[RoborockFanSpeedE2] = None
+    water_box_mode: Optional[RoborockMopIntensityV2] = None
+
+
+@dataclass
+class S6PureStatus(Status):
+    fan_power: Optional[RoborockFanSpeedS6Pure] = None
+
+
+@dataclass
+class S7MaxVStatus(Status):
+    fan_power: Optional[RoborockFanSpeedS7MaxV] = None
+    water_box_mode: Optional[RoborockMopIntensityS7] = None
+    mop_mode: Optional[RoborockMopModeS7] = None
+
+
+@dataclass
+class S7Status(Status):
+    fan_power: Optional[RoborockFanSpeedS7] = None
+    water_box_mode: Optional[RoborockMopIntensityS7] = None
+    mop_mode: Optional[RoborockMopModeS7] = None
+
+
+@dataclass
+class S8ProUltraStatus(Status):
+    fan_power: Optional[RoborockFanSpeedS7MaxV] = None
+    water_box_mode: Optional[RoborockMopIntensityS7] = None
+    mop_mode: Optional[RoborockMopModeS8ProUltra] = None
+
+
+ModelStatus: dict[str, Type[Status]] = {
```

Comment:
I think keeping this in normal dict style would be nice to make it clear.I was confused when I saw it above,, I thought it might have been some kind of generic. 
But that's just a styling thing and not really important

Reply:
yeah model_status vs ModelStatus. not a big deal though